packages/components/src/text/mixins.js

```diff
@@ -0,0 +1,150 @@
+/**
+ * External dependencies
+ */
+import css from '@emotion/css';
+/**
+ * Internal dependencies
+ */
+import { fontFamily } from './font-family';
+
+const fontWeightNormal = `font-weight: 400;`;
+const fontWeightSemibold = `font-weight: 600;`;
+
+const title = `
+  ${ fontWeightNormal }
+`;
+
+const titleLarge = `
+	font-size: 32px;
+	line-height: 40px;
+`;
+
+const titleMedium = `
+	font-size: 24px;
+	line-height: 32px;
+`;
+
+const titleSmall = `
+	font-size: 20px;
+	line-height: 28px;
+`;
+
+const subtitle = `
+	${ fontWeightSemibold }
+	font-size: 14px;
+	line-height: 20px;
+`;
+
+const subtitleLarge = `
+	font-size: 16px;
+	line-height: 24px;
+`;
+
+const subtitleSmall = `
+	font-size: 14px;
+	line-height: 20px;
+`;
+
+const body = `
+	${ fontWeightNormal }
+`;
+
+const bodyLarge = `
+	font-size: 16px;
+	line-height: 24px;
+`;
+
+const bodySmall = `
+	font-size: 14px;
+	line-height: 20px;
+`;
+
+const button = `
+  ${ fontWeightSemibold }
+  font-size: 14px;
+  line-height: 20px;
+`;
+
+const caption = `
+	${ fontWeightNormal }
+	font-size: 12px;
+	line-height: 16px;
+`;
+
+const label = `
+	${ fontWeightSemibold }
+	font-size: 12px;
+	line-height: 16px;
+`;
+
+/**
+ * @typedef {"title.large" | "title.medium" | "title.small" | "subtitle" | "subtitle.small" | "body" | "body.small" | "button" | "caption" | "label"} TextVariant
+ */
+
+/**
+	* @param {TextVariant} variant
+	*/
+// eslint-disable-next-line no-shadow
```

Comment:
Can you explain the reason for disabling this rule? We should avoid ad-hoc disabling of rules, else have a broader conversation about whether the rule provides value or whether it should be removed entirely from the ruleset. 

I think it would be easy enough to just rename the argument, no? Something like `name` or `variantName` ?